client/src/main/scala/spatutorial/client/components/QueueUserDeskRecs.scala

```diff
@@ -53,19 +55,87 @@ object QueueUserDeskRecsComponent {
             props.simulationResultPotRCP(simulationResultPotMP => {
               props.userDeskRecsPotRCP(
                 userDeskRecsPotMP => UserDeskRecsComponent(props.terminalName, props.queueName, userDeskRecsRows,
-                  props.airportInfoPotsRCP, props.flightsPotRCP, userDeskRecsPotMP, simulationResultPotMP)
+                  props.airportConfig, props.airportInfoPotsRCP, props.flightsPotRCP, userDeskRecsPotMP, simulationResultPotMP)
               )
             })),
           props.simulationResultPotRCP(simulationResultPotMP => {
             props.crunchResultPotRCP(crunchResultPotMP => {
               props.labelsPotRCP(labelsPotMP =>
                 <.div(^.cls := "user-desk-recs-chart",
-                  labelsPotMP().renderReady(labels => DeskRecsChart.userSimulationWaitTimesChart(props.terminalName, props.queueName, labels, simulationResultPotMP, crunchResultPotMP)))
+                  labelsPotMP().renderReady(labels => DeskRecsChart.userSimulationWaitTimesChart(props.terminalName, props.queueName, props.airportConfig, labels, simulationResultPotMP, crunchResultPotMP)))
               )
             })
           })
         )))
   }
 
+  def stuff() = {
+    val airportConfigPotRCP = SPACircuit.connect(_.airportConfigHolder)
+    airportConfigPotRCP(airportConfigPotMP => {
+      <.div(
+        airportConfigPotMP().renderEmpty("Hello empty"),
+        airportConfigPotMP().renderPending(_ => "Hello pending"),
+        airportConfigPotMP().renderReady((airportConfig: AirportConfigHolder) => {
+          val queueUserDeskRecProps: Seq[QueueUserDeskRecsComponent.Props] = airportConfig.terminalNames.flatMap { terminalName =>
+            airportConfig.queues.map { queueName =>
+              val labelsPotRCP = SPACircuit.connect(_.workload.map(_.labels))
+              val crunchResultPotRCP = SPACircuit.connect(_.queueCrunchResults.getOrElse(terminalName, Map()).getOrElse(queueName, Empty).flatMap(_._1))
```

Comment:
consider whether you want just one connect/zoom that get's all of this into a single, better structures RCP